monai/utils/hub_mixin.py

```diff
@@ -0,0 +1,37 @@
+from monai.utils import OptionalImportError
+
+__all__ = ["MonaiHubMixin"]
+
+
+class DummyPyTorchModelHubMixin:
+
+    error_message = "To use `{}` method please required packages: `pip install huggingface_hub safetensors`."
+
+    def __init_subclass__(cls, *args, **kwargs):
+        super().__init_subclass__()
+
+    @classmethod
+    def from_pretrained(cls, *args, **kwargs):
+        raise OptionalImportError(cls.error_message.format("from_pretrained"))
+
+    def save_pretrained(self, *args, **kwargs):
+        raise OptionalImportError(self.error_message.format("save_pretrained"))
+
+    def push_to_hub(self, *args, **kwargs):
+        raise OptionalImportError(self.error_message.format("push_to_hub"))
+
+
+try:
+    from huggingface_hub import PyTorchModelHubMixin
+except ImportError:
+    PyTorchModelHubMixin = DummyPyTorchModelHubMixin
```

Comment:
Hi @qubvel Thanks for the contribution. We have our own [optional_import](https://github.com/Project-MONAI/MONAI/blob/7449c4d23137e6fd264b35cc71c5396e2fce03c7/monai/utils/module.py#L336) function to do something similar. 

```suggestion
PyTorchModelHubMixin, has_hg_hub = optional_import("huggingface_hub", name="PyTorchModelHubMixin")
if not has_hg_hub:
    PyTorchModelHubMixin = DummyPyTorchModelHubMixin  # could put DummyPyTorchModelHubMixin definition here instead
```

Reply:
`optional_import` provides its own dummy class which raises an exception whenever a member is requested, but this doesn't have `__init_subclass__` so your version is needed.

Reply:
ok, great, I will update it!